pkg/plugins/golang/v4/init.go

```diff
@@ -187,21 +187,23 @@ func checkDir() error {
 			if isCapitalized && info.Name() != "PROJECT" {
 				return nil
 			}
-			// Allow files in the following list
-			allowedFiles := []string{
-				"go.mod", // user might run `go mod init` instead of providing the `--flag` at init
-				"go.sum", // auto-generated file related to go.mod
+			// Deny files with .go or .yaml extensions
+			disallowedExtensions := []string{
```

Comment:
this changes behavior of accepting having `go.mod` and `go.sum` files when creating . Is this expected?

Reply:
do you think creating a separate logic for this to handle will work? like 
`disallowedFiles := []string{"go.mod", "go.sum"}`